website/bot.py

```diff
@@ -0,0 +1,126 @@
+from pathlib import Path
+
+import openai
+from dotenv import find_dotenv, load_dotenv
+from langchain.chains import ConversationalRetrievalChain
+from langchain.memory import ConversationSummaryMemory
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_community.document_loaders import (
+    DirectoryLoader,
+    Docx2txtLoader,
+    PyPDFLoader,
+    TextLoader,
+    UnstructuredMarkdownLoader,
+)
+from langchain_community.vectorstores import FAISS
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI, OpenAIEmbeddings
+from openai import OpenAI
+
+load_dotenv(find_dotenv(), override=True)
+
+
+def is_api_key_valid(api_key):
+    client = OpenAI(api_key=api_key)
+    try:
+        response = client.completions.create(
+            prompt="Hello", model="gpt-3.5-turbo-instruct", max_tokens=1
+        )
+        return True
+    except openai.APIError as e:
+        return f"OpenAI API returned an API Error: {e}"
+    except openai.APIConnectionError as e:
+        return f"Failed to connect to OpenAI API: {e}"
+    except openai.RateLimitError as e:
+        return f"OpenAI API request exceeded rate limit: {e}"
```

Comment:
`openai.APIConnectionError` and `openai.RateLimitError` will never be reached as `openai.APIError` is their base class. You need to put it last.

Also you don't need to return the error messages. The current `is_api_key_valid()` function result is always truthy in case of openai error happening.

Reply:
now I added print for find the error by developer when they are doing to debug in future